File: 2405-minimum-number-of-keypresses/2405-minimum-number-of-keypresses.java
class Solution {
    public int minimumKeypresses(String s) {
        // Create array of all char to store its occurences
        int[] seq = new int[26];

        for(char c: s.toCharArray()){
            seq[c - 'a']++;
        }
        // Sort the array
        Arrays.sort(seq);
        
        int res =0;

        for(int i=25; i >=0; i--){
            if(i > 16 ){
                res += seq[i];  // For first chars in keyboard
            } else if (i > 7 && i <=16){
                res += seq[i]*2; // For second chars in keyboard
            } else {
                res += seq[i]*3; // For third chars in keyboard
            }
        }

        return res;
    }
}

File: 2405-minimum-number-of-keypresses/2405-minimum-number-of-keypresses.py
class Solution:
    def minimumKeypresses(self, s: str) -> int:
        res = 0 

        # To create counter of each char
        cnts = collections.Counter(s)
        print(cnts)

        # Sort the Counter
        cnts = sorted(cnts.items(), key=lambda x: -x[1])
        print(cnts)


        for idx,(char, cnt) in enumerate(cnts):
            # For First chars in keyboard
            if idx <=8:
                res += cnt
            # For second chars in keyboard
            elif idx > 8 and idx <= 17:
                res += cnt*2
            else:
                res += cnt*3

        return res
